include/xten/Dialect/XTenNN/IR/XTenNNOps.td

```diff
@@ -298,6 +298,31 @@ def XtenNN_CosOp: XTenNN_Op<"cos", [Pure, TosaExtension, ElementwiseUnary, SameO
   let assemblyFormat = [{ operands attr-dict `:` functional-type(operands, results) }];
 }
 
+
+def XtenNN_DepthToSpaceOp: XTenNN_Op<"depth_to_space", [Pure, TosaExtension]> {
+  let summary = "Performs resizing to an image.";
+  let description = [{
+    Permutes data from depth into blocks of spatial data.
+    Outputs a copy of the input tensor where values from the depth dim are moved to the H and W dimension.
+    Blocksize: The size of the blocks being moved.
+    Mode:
+    - 1: DCR: Elements are rearranged in order Depth - Column - Row
+    - 2: CRD: Elements rearranged in order Column - Row - Depth
+  }];
+  let arguments = (ins
+    4DTensorOf<[BF16, F32]>:$X,
+    I64Attr:$blocksize,
+    I64Attr:$mode
```

Comment:
do we support all modes? Should we have a verifier?

Reply:
We support all modes, I tried writing a verifier but it already fails in ONNX MLIR if the mode is invalid.